core/services/gateway/handlers/functions/handler.functions.go

```diff
@@ -48,10 +104,21 @@ func (h *functionsHandler) HandleNodeMessage(ctx context.Context, msg *api.Messa
 	return nil
 }
 
-func (h *functionsHandler) Start(context.Context) error {
+func (h *functionsHandler) Start(ctx context.Context) error {
+	if h.allowlist != nil {
+		checkFreq := time.Duration(h.handlerConfig.AllowlistUpdateFrequencySec) * time.Second
+		checkTimeout := time.Duration(h.handlerConfig.AllowlistUpdateTimeoutSec) * time.Second
+		h.shutdownWaitGroup.Add(1)
+		go func() {
+			h.allowlist.UpdatePeriodically(h.serviceContext, checkFreq, checkTimeout)
+			h.shutdownWaitGroup.Done()
+		}()
+	}
 	return nil
 }
 
 func (h *functionsHandler) Close() error {
+	h.serviceCancel()
+	h.shutdownWaitGroup.Wait()
```

Comment:
The second call to Close will block forever. This is why I mentioned StartStopOnce util.